### DISCUSSION TITLE
About the executing user of the task instance

### DISCUSSION THREAD

**Discussion by STTianXiang** [apache/dolphinscheduler#10101] [Q&A]
Due to historical reasons, we have more users. Now we are going to divide the task into a user, so the execution user in the "workflow instance" needs to be unified. But I don't know how to modify them in batches. How can I change the execution user in the picture below to a unified user? Looking forward and thanks for your reply!

![微信图片_20220518173053](https://user-images.githubusercontent.com/50575666/169007115-de1bff6e-da44-4cca-a44f-a36d6b0ee98f.jpg)

**Comment by Tandoy** [comment]
If you only need to modify the execution user of the workflow instance display page, you can modify the metadata in the metabase
t_ds_process_instance of executor_id field

https://dolphinscheduler.apache.org/zh-cn/docs/latest/user_doc/architecture/metadata.html

**Reply by STTianXiang** [reply]
Does the executor_id field correspond to the id field of the t_ds_user table?

**Reply by Tandoy** [reply]
> Does the executor_id field correspond to the id field of the t_ds_user table?

You may need to look at the data to determine

**Reply by STTianXiang** [reply]
What I understand is that the instances in the t_ds_process_instance table are generated when the task runs. Where does his executive user get it from? Because what I want is to execute the user uniformly before the task runs, rather than modifying them after the task runs.

**Reply by Tandoy** [reply]
> What I understand is that the instances in the t_ds_process_instance table are generated when the task runs. Where does his executive user get it from? Because what I want is to execute the user uniformly before the task runs, rather than modifying them after the task runs.

The default is the login user. If unified users are required, they may need to be represented
